src/test/java/MacroExecutionModuleImplementationTest.java

```diff
@@ -0,0 +1,236 @@
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.Ignore;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public final class MacroExecutionModuleImplementationTest {
+    private static final String MACRO_NAME = "TestBot";
+    private static final String BOT_NAME = "@MacroBot";
+    private static final String ROOM1 = "spaces/AAAAAAAAAAA";
+    private static final String THREAD1 = "spaces/AAAAAAAAAAA/threads/BBBBBBBBBBB";
+    private static final String USER_EMAIL1 = "testemail1@google.com";
+    private static final String USER_EMAIL2 = "testemail2@google.com";
+    private static final String MESSAGE_WITH_MACRO_NAME = String.format("@MacroBot %s this is my message", MACRO_NAME);
+    private static final String MESSAGE_WITHOUT_MACRO_NAME = "@MacroBot this is my message";
+    private static final String MESSAGE_CONTENT_ONLY = "this is my message";
+    private static final String EMPTY_MESSAGE_RESPONSE = "You must type a message when you message me. Please type \"@MacroBot /help\" for more instructions.";
+    private static final String HELP_MESSAGE_REPLY = "This is your help message";
+    private static final String HELP_MESSAGE = "@MacroBot /help";
+    private static final String NO_ACCESS_MESSAGE = String.format("You do not own/have access to %s.", MACRO_NAME);
+    private static final String SHARE_MESSAGE_REPLY = String.format("The %s macro belonging to %s has been shared to this room. All users in this room can use this macro.", MACRO_NAME, USER_EMAIL1);
+    private static final String SHARE_MESSAGE = String.format("@MacroBot /share %s", MACRO_NAME);
+    private static final String SHEET_URL = "d/sheetId";
+    private static final String SHEET_ID = "sheetId";
+
+    private static SheetEntryType[] columnHeaders = {SheetEntryType.TIME, SheetEntryType.EMAIL, SheetEntryType.CONTENT};
+    private static final Macro DEFAULT_MACRO = new Macro(USER_EMAIL1, MACRO_NAME, new CommandTrigger("Command Trigger"), new SheetAppendRowAction(SHEET_URL, columnHeaders)); 
+
+    private static MacroExecutionModuleImplementation execution;
+    private static EntityModule mockEntityModule;
+    private static CloudDocClient mockCloudDocClient;
+    private static CloudSheet mockCloudSheet;
+
+    @BeforeClass
+    public static void setUpOnce() {
+        mockEntityModule = Mockito.mock(EntityModule.class);
+        mockCloudDocClient = Mockito.mock(CloudDocClient.class);
+        mockCloudSheet = Mockito.mock(CloudSheet.class);   
+        execution  = MacroExecutionModuleImplementation.initializeServer(mockEntityModule, mockCloudDocClient);
+    }
+
+    @Before
+    public void setUpForEachTest() {
+        execution.getThreadMacroMap().clear();
+        execution.getRoomToMacro().clear();
+    }
+    
+    // Remove a macro from the roomToMacro Map.
+    @Test
+    public void testRemoveMacro() {
+        shareDefaultMacroToRoom(ROOM1);
+
+        execution.removeMacro(ROOM1);
```

Comment:
nit: so here is an example of a place in which it would be good to assert the macro got indeed stored. Otherwise the final assert could pass even without calling execute.